objax/functional/loss.py

```diff
@@ -38,6 +40,8 @@ def cross_entropy_logits(logits: JaxArray, labels: JaxArray) -> JaxArray:
     Returns:
         (batch, ...) tensor of the cross-entropies for each entry.
     """
+    if logits.shape != labels.shape:
+        warnings.warn(' {} {} : arg1 {} and arg2 {}'.format(warn_mess, __all__[0], logits.shape, labels.shape))
```

Comment:
Using `__all__[index]` is very brittle to refer to function name.
If somebody later change order of functions in `__all__` then all warnings will be broken.
The simple way is to just explicitly write function name as a string: `warningns.warn(' .... '. format(WARN_MESS, 'cross_entropy_logits', logits.shape, labels.shape)`

You can look into programmatically getting function name from within a function.
The best option I found is `inspect.currentframe().f_code.co_name`, but it's not particularly nice or readable.
See here for some discussion: https://stackoverflow.com/questions/5067604/determine-function-name-from-within-that-function-without-using-traceback